### ISSUE TITLE
[Azure Devops Pipeline]: Error re-running test with TestCase Attribute

### ISSUE THREAD

**Issue by alkampfergit** [nunit/nunit3-vs-adapter#817]
I have an Azure DevOps pipeline with test runner configured to re-run failed test. If the test that failed on the run is a test with TestCase Attribute, it is not possible to re-run the test due to this error.

`##[error]An exception occurred while invoking executor 'executor://nunit3testexecutor/': Filter string '"v1"\,True\,"C2"' includes unrecognized escape sequence.
`

This is the command line that got generated, as you can see the fully qualified name of the test contains parameters of TestCase, and this will break command line.

```
vstest.console.exe /TestCaseFilter:"FullyQualifiedName=Jarvis.InProcessIntegration.Scenarios.ProductCatalog.QuoteBom.QuoteBomPriceListClassTests.Check_feature_flag_to_have_accessories_inherits_price_list_class(\"v1\"\,True\,\"C2\")"
"C:\A\_work\1402\s\src\Tests\InProcessIntegration\Jarvis.InProcessIntegration.Scenarios\bin\Debug\net461\Jarvis.InProcessIntegration.Scenarios.dll"
"C:\A\_work\1402\s\src\Tests\InProcessIntegration\Jarvis.InProcessIntegration.Scenarios\obj\Debug\net461\Jarvis.InProcessIntegration.Scenarios.dll"
/Settings:"C:\A\_work\_temp\w3mloqdr2u5.tmp.runsettings"
```

This is annoying if you have some flacky integration test, because it will break re-run failed test option.

**Comment by OsirisTerje** [comment]
I'll have a look, Gian .   
Does it happens regardless of the parameter type?

**Comment by alkampfergit** [comment]
I'm investigating but surely it happens with string and char paramters. Here is one test that causes the problem.
![image](https://user-images.githubusercontent.com/358545/107008198-cb15f780-6793-11eb-87b1-340c6d866949.png)

This is a flacky test, we have some of them inside our test base. This is the name of the test as in pipeline output

![image](https://user-images.githubusercontent.com/358545/107008328-f1d42e00-6793-11eb-9c4c-d7ddf367527b.png)

Clearly it has a (',') in the name that is strange. Once it try re-run VSTask is issuing this command

```
vstest.console.exe /TestCaseFilter:"FullyQualifiedName=Jarvis.Tests.Support.DomainAssumptionTestsCommands.Verify_that_there_is_no_command_with_wrong_getter|FullyQualifiedName=Jarvis.Common.Shared.Tests.Search.OmniSearchQueryClientTests.Logging_for_query_error|FullyQualifiedName=Jarvis.Common.Shared.Tests.Forms.JarvisFormDataLookupCsvImporterTests.Import_base_file('\,')"
```

As you can see at the end of the commandline, it uses FullyQualifiedName but the name that it is passing is wrong because contains parenthesis etc. If I have time I'll try to replicate in a simple project in a public project so you can investigate immediately to the issue.

Gian Maria.

**Comment by alkampfergit** [comment]
Hi Terje,

to simplify your investigation, here is a repository with a build that can reproduce the problem [https://github.com/alkampfergit/NunitTestReRunProblem](https://github.com/alkampfergit/NunitTestReRunProblem)

I've [Got a build on a public project of mine](https://dev.azure.com/gianmariaricci/Public/_build/results?buildId=3287&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=7bfb7e0f-549d-5929-c603-6e8046613b9e) but you can simply run on your account with no problem, it will generate the very same problem I had on my real project. 

Basically when we are using test case the adapter generates test names, that contains parenthesis and other chars, and when they are specified as FullyQualifiedName by the test runner to re-run only failed test it will make command line parser fail because it probably expects real valid method names, not containing any weird chars.

![image](https://user-images.githubusercontent.com/358545/107010359-8770bd00-6796-11eb-9976-d30b5aaead07.png)

**Comment by alkampfergit** [comment]
@OsirisTerje any news?

**Comment by OsirisTerje** [comment]
This is part of the FQN issues.  We're waiting for MS to fix their part of this before we can continue.  I've tried to rework the example by setting the options to use parents for FQN, but VS hangs on that now.

**Comment by alkampfergit** [comment]
Thanks.

**Comment by OsirisTerje** [comment]
@alkampfergit Can you check this again using the latest VS,  version 16.11.3, components ?  It seems they have fixed a bunch of these errors.

**Comment by OsirisTerje** [comment]
Awesome!   Then we can close this one too.

**Comment by MaxB504** [comment]
Hello.  I have upgraded VS Enterprise (to 16.11.5) on our image for our self hosted VMs and still encounter this error with test re-runs.  I used Visual Studio Installer to perform the upgrade.  Is there another way I should upgrade specific components?
The Command Line Tool Version is 16.11.0.

**Comment by zsbolgar** [comment]
Hi, 
I have same issue.

- Microsoft.TestPlatform 17.3.0-preview-20220426-02
- NUnit3.TestAdapter : 4.2.0.0

NUnit Adapter 4.2.0.0: Test execution started
##[error]An exception occurred while invoking executor 'executor://nunit3testexecutor/': Filter string '"repo64char_123456789_123456789_123456789_123456789_123456789_123"\,"ED50-UTM31"\,null' includes unrecognized escape sequence.
##[error]Stack trace:

**Comment by ciconq** [comment]
I can confirm I am getting the same exception as @zsbolgar:

`##[error]An exception occurred while invoking executor 'executor://nunit3testexecutor/': Filter string '"firstStringValue"\,"secondStringValue"\,"thirdStringValue"' includes unrecognized escape sequence.`

I am using:

* NUnit 4.1.0
* NUnit3TestAdapter 4.5.0
